tests/testthat/test-shinytest2-tm_data_table.R

```diff
@@ -0,0 +1,41 @@
+test_that("e2e: tm_front_page initializes without errors", {
+  skip_if_too_deep(5)
+  require(shinytest2)
+
+  app <- TealAppDriver$new(
+    data = simple_teal_data(),
+    modules = teal::modules(
+      tm_data_table(
+        variables_selected = list(
+          iris = c("Sepal.Length", "Sepal.Width", "Petal.Length", "Petal.Width", "Species")
+        ),
+        datasets_selected = "mtcars",
+        dt_args = list(caption = "Table Caption")
+      )
+    ),
+    timeout = 3000
+  )
+
+  app$expect_no_shiny_error()
+
+  # tables
+  app$expect_screenshot(selector = app$active_module_element("mtcars-data_table"))
```

Comment:
@kartikeyakirar what does `$expect_screenshot()` do? Should it compare to some pre-defined screenshot?

Reply:
As per my understanding from document shinytest2 captures screenshots of the app's UI at points specified by selector in  app$expect_screenshot(). These initial screenshots are stored as reference snapshots. we can also defined specified path for comparison 
![image](https://github.com/insightsengineering/teal.modules.general/assets/6700955/3dce29af-35aa-4a1a-ab1b-64193c9bdfd3)

Reply:
so it takes the snapshot and then what : P ?

Reply:
The function captures a snapshot and either stores it in the _snap directory or compares it with an existing snapshot already stored in that directory.

Reply:
anyways I am updating the PR and removing screenshots 😄

Reply:
ok, so first time it takes the snapshot. next time it compares to the previous one. thanks